core/src/main/java/org/ldaptive/dn/RDn.java

```diff
@@ -0,0 +1,212 @@
+/* See LICENSE for licensing and NOTICE for copyright. */
+package org.ldaptive.dn;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.ldaptive.LdapUtils;
+
+/**
+ * Relative distinguished name containing one or more name value pairs. Name value pairs are ordered from left to right
+ * such that the left-most pair is considered the first. For the RDN 'cn=Jane Doe+mail=jdoe@example.com', the first name
+ * value pair is 'cn=Jane Doe'.
+ *
+ * See <a href="http://www.ietf.org/rfc/rfc4514.txt">RFC 4514</a> for more details on the string representations of
+ * RDNs.
+ *
+ * @author  Middleware Services
+ */
+public class RDn
+{
+
+  /** hash code seed. */
+  private static final int HASH_CODE_SEED = 5009;
+
+  /** Name value pairs. */
+  private final Set<NameValue> nameValues;
+
+
+  /**
+   * Creates a new RDN with the supplied string.
+   *
+   * @param  rdn  to parse
+   *
+   * @throws  IllegalArgumentException  if rdn contains multiple RDNs or no RDNs
+   */
+  public RDn(final String rdn)
+  {
+    this(rdn, new DefaultDnParser());
+  }
+
+
+  /**
+   * Creates a new RDN with the supplied string.
+   *
+   * @param  rdn  to parse
+   * @param  parser  to parse dn
+   *
+   * @throws  IllegalArgumentException  if rdn contains multiple RDNs or no RDNS
+   */
+  public RDn(final String rdn, final DnParser parser)
+  {
+    final List<RDn> rdns = parser.parse(rdn);
+    if (rdns.isEmpty()) {
+      throw new IllegalArgumentException("Invalid RDN: no RDNs found in " + rdn);
+    }
+    if (rdns.size() > 1) {
+      throw new IllegalArgumentException("Invalid RDN: multiple RDNs found in " + rdn);
+    }
+    nameValues = rdns.get(0).getNameValues();
+  }
+
+
+  /**
+   * Creates a new RDN with the supplied name value pairs.
+   *
+   * @param  value  to add
+   */
+  public RDn(final NameValue... value)
+  {
+    nameValues = Stream.of(value).filter(Objects::nonNull).collect(
+      Collectors.collectingAndThen(Collectors.toCollection(LinkedHashSet::new), Collections::unmodifiableSet));
+  }
+
+
+  /**
+   * Creates a new RDN with the supplied name value pairs.
+   *
+   * @param  values  to add
+   */
+  public RDn(final Collection<NameValue> values)
+  {
+    nameValues = values.stream().filter(Objects::nonNull).collect(
+      Collectors.collectingAndThen(Collectors.toCollection(LinkedHashSet::new), Collections::unmodifiableSet));
+  }
+
+
+  /**
+   * Creates a new RDN with a single name value pair.
+   *
+   * @param  attributeName  to add
+   * @param  attributeValue  to add
+   */
+  public RDn(final String attributeName, final String attributeValue)
+  {
+    nameValues = Set.of(new NameValue(attributeName, attributeValue));
```

Comment:
I believe `Set.of` returns an immutable list, but I wanted to call it out to double check given behavior of other constructors.

Reply:
Correct. `Set.of` is immutable.